### PR TITLE
Adding test_coding_standards.py + py.test

### PR DESCRIPTION
**Pull request by ocefpaf**
Adapted `test_coding_standards.py` from iris.

### CODE REVIEW

File: cf_units/tests/test_coding_standards.py [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+LICENSE_RE_PATTERN = re.escape(LICENSE_TEMPLATE).replace('\{YEARS\}', '(.*?)')
+# Add shebang possibility to the LICENSE_RE_PATTERN
+LICENSE_RE_PATTERN = r'(\#\!.*\n)?' + LICENSE_RE_PATTERN
+LICENSE_RE = re.compile(LICENSE_RE_PATTERN, re.MULTILINE)
+
+
+# Guess cf_units repo directory of cf_units - realpath is used to mitigate
+# against Python finding the cf_units package via a symlink.
+IRIS_DIR = os.path.realpath(os.path.dirname(cf_units.__file__))
+REPO_DIR = os.path.dirname(os.path.dirname(IRIS_DIR))
```
Comment (rhattersley) [inline cf_units/tests/test_coding_standards.py]: A couple of left-over "IRIS" references.

File: cf_units/tests/test_coding_standards.py [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+    def whatchanged_parse(whatchanged_output):
+        """
+        Returns a generator of tuples of data parsed from
+        "git whatchanged --pretty='TIME:%at". The tuples are of the form
+        ``(filename, last_commit_datetime)``
+
+        Sample input::
+
+            ['TIME:1366884020', '',
+             ':000000 100644 0000000... 5862ced... A\tlib/iris/cube.py']
```
Comment (rhattersley) [inline cf_units/tests/test_coding_standards.py]: Another "iris" reference that would be nice to get rid of.

**Comment by rhattersley** [comment]
The test report says "127 passed, 1 skipped in 0.80 seconds" ... I'm not sure why 1 has been skipped. Looking at a [previous test report](https://travis-ci.org/SciTools/cf_units/builds/61447175) there aren't any skipped.

It'd be helpful to run the tests in verbose mode to see what's going on.

**Comment by ocefpaf** [comment]
Thanks @rhattersley!  I know this is time consuming for you, but I am learning a lot!

I removed all remaining iris references.

> I'm not sure why 1 has been skipped. Looking at a previous test report there aren't any skipped.

The skipped test is `test_license_headers`.  (See https://travis-ci.org/SciTools/cf_units/builds/61745633#L341.) I will check what is going on later today.

**Comment by ocefpaf** [comment]
@rhattersley I added `test_Unit.py`, but left `test_suppress_unit_warnings` out for now.  I am unsure how to implement this part: https://github.com/SciTools/iris/blob/master/lib/iris/tests/unit/unit/test_suppress_unit_warnings.py#L51

**Comment by rhattersley** [comment]
> I am unsure how to implement this part: https://github.com/SciTools/iris/blob/master/lib/iris/tests/unit/unit/test_suppress_unit_warnings.py#L51

The warning catching behaviour of `suppress_unit_warnings` is specific to Iris, so should be removed. I've sent you ocefpaf/cf_units#1 which does just that.

I'm not sure how to test the remaining error suppression behaviour because I'm not sure how to get UDUNITS-2 to spit out an error message (other than on startup).

**Comment by ocefpaf** [comment]
Thanks for the PR!

> I'm not sure how to get UDUNITS-2 to spit out an error message (other than on startup).

I will take a look at this.

**Comment by ocefpaf** [comment]
Rebased and (hopefully) good to go :wink:

**Comment by rhattersley** [comment]
Super :+1:

**Pull request merged**